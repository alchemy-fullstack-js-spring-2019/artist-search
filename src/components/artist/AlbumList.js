import React from 'react';
import PropTypes from 'prop-types';
import Album from './Album';
import { Link } from 'react-router-dom';

function AlbumList({ albumArr, artist }) {
  const allAlbums = albumArr.map(album => {
    return (
      <Link key={album.id} to={`/album/${album.id}/${artist}/${album.title}`}>
        <li style={{ margin: '20px', background: 'lavender', padding: '10px', color: 'black' }}>
          <Album album={album} />
        </li>
      </Link>
    );
  });

  return (
    <ul style={{ display: 'flex', flexWrap: 'wrap', width: '100%', listStyle: 'none' }}>
      {allAlbums}
    </ul>
  );
}

AlbumList.propTypes = {
  albumArr: PropTypes.array.isRequired,
  artist: PropTypes.string.isRequired
};

export default AlbumList;
