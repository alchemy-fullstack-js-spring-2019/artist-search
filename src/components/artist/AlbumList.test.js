import React from 'react';
import { shallow } from 'enzyme';
import AlbumList from './AlbumList';

describe('AlbumList component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<AlbumList albumArr={[]} artist='' />);
    expect(wrapper).toMatchSnapshot();
  });
});
