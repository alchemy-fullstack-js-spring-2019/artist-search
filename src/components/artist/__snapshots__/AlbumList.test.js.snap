// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumList component matches snapshot 1`] = `
<ul
  style={
    Object {
      "display": "flex",
      "flexWrap": "wrap",
      "listStyle": "none",
      "width": "100%",
    }
  }
/>
`;
